File: backend/app/schemas/fact_sheet.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.fact_sheet import FactSheetStatus, FactSheetType, QualitySeal


class FactSheetBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=500)
    type: FactSheetType
    description: str | None = None
    display_name: str | None = None
    alias: str | None = None
    external_id: str | None = None
    parent_id: uuid.UUID | None = None
    lifecycle: dict | None = None
    attributes: dict | None = None


class FactSheetCreate(FactSheetBase):
    pass


class FactSheetUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=500)
    description: str | None = None
    display_name: str | None = None
    alias: str | None = None
    status: FactSheetStatus | None = None
    external_id: str | None = None
    parent_id: uuid.UUID | None = None
    quality_seal: QualitySeal | None = None
    lifecycle: dict | None = None
    attributes: dict | None = None


class FactSheetRead(FactSheetBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    status: FactSheetStatus
    quality_seal: QualitySeal | None
    completion: float
    created_at: datetime
    updated_at: datetime


class FactSheetList(BaseModel):
    items: list[FactSheetRead]
    total: int
    page: int
    page_size: int

File: frontend/src/types/fact-sheet.ts
export type FactSheetType =
  | "application"
  | "business_capability"
  | "business_context"
  | "organization"
  | "objective"
  | "it_component"
  | "tech_category"
  | "provider"
  | "interface"
  | "data_object"
  | "initiative"
  | "platform";

export type FactSheetStatus = "active" | "archived";

export interface FactSheet {
  id: string;
  name: string;
  display_name: string | null;
  description: string | null;
  type: FactSheetType;
  status: FactSheetStatus;
  alias: string | null;
  external_id: string | null;
  parent_id: string | null;
  quality_seal: string | null;
  completion: number;
  lifecycle: Record<string, string> | null;
  attributes: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
}

export interface FactSheetListResponse {
  items: FactSheet[];
  total: number;
  page: number;
  page_size: number;
}

export const FACT_SHEET_TYPE_LABELS: Record<FactSheetType, string> = {
  application: "Application",
  business_capability: "Business Capability",
  business_context: "Business Context",
  organization: "Organization",
  objective: "Objective",
  it_component: "IT Component",
  tech_category: "Tech Category",
  provider: "Provider",
  interface: "Interface",
  data_object: "Data Object",
  initiative: "Initiative",
  platform: "Platform",
};

export const FACT_SHEET_TYPE_ICONS: Record<FactSheetType, string> = {
  application: "apps",
  business_capability: "account_tree",
  business_context: "category",
  organization: "corporate_fare",
  objective: "flag",
  it_component: "memory",
  tech_category: "layers",
  provider: "storefront",
  interface: "swap_horiz",
  data_object: "database",
  initiative: "rocket_launch",
  platform: "hub",
};
